start

```diff
@@ -491,15 +493,71 @@ function stop_postgres() {
   while kill -0 "$CURRENT_POSTGRES_PID" &> /dev/null; do
     sleep 0.5
   done
-  echo "postgres: down"
+  print_output "postgres: down"
 }
 
 pushd () {
-    command pushd "$@" > /dev/null
+  command pushd "$@" > /dev/null
 }
 
 popd () {
-    command popd "$@" > /dev/null
+  command popd "$@" > /dev/null
+}
+
+function service_status () {
+  horizon_status
+  if [ "$ENABLE_SOROBAN_RPC" = "true" ]; then
+    soroban_rpc_status
+  fi   
+}
+
+function soroban_rpc_status () { 
+  print_screen_output "waiting for soroban rpc to report ready state..."
+  COUNTER=1
+  while ! $(curl --silent --location --request POST 'http://localhost:8000/soroban/rpc' \
+    --header 'Content-Type: application/json' \
+    --data-raw '{
+        "jsonrpc": "2.0",
+        "id": 10235,
+        "method": "getHealth"
+        
+    }' | jq --exit-status '.result.status == "healthy"' 2>/dev/null | grep -o true || echo false ); 
+  do
+    if [ $COUNTER -gt 12 ]; then 
+      print_screen_output "waiting for soroban rpc to report ready state..."
+      COUNTER=1
+    else 
+      COUNTER=$[$COUNTER +1]    
+    fi 
+    sleep 5
+  done 
+  print_screen_output "soroban rpc reported ready status, the service can be used by tools/cli now ..."
+}
+
+function horizon_status () {
+  COUNTER=1
+  print_screen_output "waiting for horizon ingestion to be caught up to network current ledger..."
+  while ! $(curl --silent --location --request GET 'http://localhost:8000' | \
+    jq --exit-status '.core_latest_ledger > 5 and .history_latest_ledger > 5 and (.core_latest_ledger - .history_latest_ledger) < 5' 2>/dev/null | \
+    grep -o true || echo false ); 
+  do
+    if [ $COUNTER -gt 12 ]; then 
+      print_screen_output "waiting for horizon ingestion to be caught up to network current ledger..."
+      COUNTER=1
+    else 
+      COUNTER=$[$COUNTER +1] 
+    fi 
+    sleep 5
+  done 
+  print_screen_output "horizon ingestion is caught up to network current ledger ..."
+}
+
+function print_screen_output() {
+  echo "$1" | tee -a $QUICKSTART_LOG
+}
+
+function print_output() {
+  echo "$1" >> "$QUICKSTART_LOG" 
```

Comment:
Why do some things that used to be printed to screen now only printed to the quickstart log? Could we keep the things that were already printed to screen, still on screen? E.g. things like postgres coming up, etc are useful to see.

Reply:
I think we're over optimizing at the expense of removing some useful information when debugging. One of those lines outputs a password someone might need. Other lines output when work is being skipped that would normally happen. For the average users it is probably removing 1-5 lines of logs, which isn't noisy.

Reply:
reverted that behavior back to original console output. https://github.com/stellar/quickstart/pull/403/commits/46ec1a2c81085d461d9e4cead772a5e3ae12c7cb